### PR TITLE
Update to E+ v9.4.0-RC2

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Update to E+ v9.4.0-RC1
Update to E+ v9.4.0-RC2

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
Not sure why CI cannot find Openstudio.hxx. Locally I hit errors with E+ segfault on Daylighting_School.idf simulation:

```
[  0%][8/1357] Updating EnergyPlus simulation in /home/julien/Software/Others/OS-build-release/resources/energyplus/Daylighting_School/, this may take a while
FAILED: resources/energyplus/Daylighting_School/eplusout.err 
cd /home/julien/Software/Others/OS-build-release/resources && /usr/local/bin/cmake -E copy energyplus/Daylighting_School/in.idf energyplus/Daylighting_School/in.idf && /usr/local/bin/cmake -E copy /home/julien/Software/Others/OS-build-release/EnergyPlus-9.4.0-5a40ebd3b9-Linux-Ubuntu18.04-x86_64/Energy+.idd energyplus/Daylighting_School/Energy+.idd && /usr/local/bin/cmake -E copy /home/julien/Software/Others/OS-build-release/EnergyPlus-9.4.0-5a40ebd3b9-Linux-Ubuntu18.04-x86_64/WeatherData/USA_CO_Golden-NREL.724666_TMY3.epw energyplus/Daylighting_School/in.epw && /usr/local/bin/cmake -E chdir energyplus/Daylighting_School /home/julien/Software/Others/OS-build-release/EnergyPlus-9.4.0-5a40ebd3b9-Linux-Ubuntu18.04-x86_64/energyplus > energyplus/Daylighting_School/screen.out
Segmentation fault
```

I built E+ v9.4.0-RC1 in Debug mode:

```
(py38)julien@Daylighting_School$ $ep_build/Products/energyplus
EnergyPlus Starting
EnergyPlus, Version 9.4.0-5a40ebd3b9, YMD=2020.09.21 12:02
Initializing Response Factors
Calculating CTFs for "INTERIORFURNISHINGS"
Calculating CTFs for "IEAD R-15 CI ROOF"
Calculating CTFs for "INTERIOR WALL"
Calculating CTFs for "MASS R-7.6 CI EXTERIOR WALL"
Calculating CTFs for "UNHEATED 8IN EXTERIOR SLAB W/ CARPET"
Calculating CTFs for "LIGHT_SHELF_CONSTRUCTION"
Initializing Window Optical Properties
Initializing Light Shelf Daylighting Devices
Initializing Solar Calculations
Allocate Solar Module Arrays
Initializing Zone and Enclosure Report Variables
Initializing Surface (Shading) Report Variables
Determining Shadowing Combinations
Computing Window Shade Absorption Factors
energyplus: /home/julien/Software/Others/EnergyPlus/third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh:1172: T& ObjexxFCL::Array1< <template-parameter-1-1> >::operator()(int) [with T = EnergyPlus::Construction::ConstructionProps]: Assertion `contains( i )' failed.
Aborted (core dumped)
```

```
energyplus: /home/julien/Software/Others/EnergyPlus/third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh:1172: T& ObjexxFCL::Array1< <template-parameter-1-1> >::operator()(int) [with T = EnergyPlus::Construction::ConstructionProps]: Assertion `contains( i )' failed.
Process 29861 stopped
* thread #1, name = 'energyplus', stop reason = hit program assert
    frame #4: 0x00007ffff3dc4deb libenergyplusapi.so.9.4.0`ObjexxFCL::Array1<EnergyPlus::Construction::ConstructionProps>::operator(this=0x00007ffff7d4ae88, i=0)(int) at Array1.hh:1172
   1169		T &
   1170		operator ()( int const i )
   1171		{
-> 1172			assert( contains( i ) );
   1173			return sdata_[ i ];
   1174		}
   1175	
(lldb) bt
* thread #1, name = 'energyplus', stop reason = hit program assert
    frame #0: 0x00007ffff1499f47 libc.so.6`__GI_raise(sig=<unavailable>) at raise.c:51
    frame #1: 0x00007ffff149b8b1 libc.so.6`__GI_abort at abort.c:79
    frame #2: 0x00007ffff148b42a libc.so.6`__assert_fail_base(fmt="%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion="contains( i )", file="/home/julien/Software/Others/EnergyPlus/third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh", line=1172, function="T& ObjexxFCL::Array1< <template-parameter-1-1> >::operator()(int) [with T = EnergyPlus::Construction::ConstructionProps]") at assert.c:92
    frame #3: 0x00007ffff148b4a2 libc.so.6`__GI___assert_fail(assertion=<unavailable>, file=<unavailable>, line=<unavailable>, function=<unavailable>) at assert.c:101
  * frame #4: 0x00007ffff3dc4deb libenergyplusapi.so.9.4.0`ObjexxFCL::Array1<EnergyPlus::Construction::ConstructionProps>::operator(this=0x00007ffff7d4ae88, i=0)(int) at Array1.hh:1172
    frame #5: 0x00007ffff4adff91 libenergyplusapi.so.9.4.0`EnergyPlus::SolarShading::ComputeWinShadeAbsorpFactors() at SolarShading.cc:11800
    frame #6: 0x00007ffff4a5c542 libenergyplusapi.so.9.4.0`EnergyPlus::SolarShading::InitSolarCalculations(ioFiles=0x00007fffffff8550) at SolarShading.cc:444
    frame #7: 0x00007ffff5ce5cc5 libenergyplusapi.so.9.4.0`EnergyPlus::HeatBalanceManager::InitHeatBalance(state=0x00007fffffff70d0, dataWindowComplexManager=0x00007fffffff9398, dataWindowEquivalentLayer=0x00007fffffff9508, dataWindowManager=0x00007fffffff9640, ioFiles=0x00007fffffff8550) at HeatBalanceManager.cc:5207
    frame #8: 0x00007ffff5ca2823 libenergyplusapi.so.9.4.0`EnergyPlus::HeatBalanceManager::ManageHeatBalance(state=0x00007fffffff70d0) at HeatBalanceManager.cc:372
    frame #9: 0x00007ffff4a23a51 libenergyplusapi.so.9.4.0`EnergyPlus::SizingManager::SetupZoneSizing(state=0x00007fffffff70d0, ErrorsFound=0x00007ffff7d74009) at SizingManager.cc:3954
    frame #10: 0x00007ffff49f9a4a libenergyplusapi.so.9.4.0`EnergyPlus::SizingManager::ManageSizing(state=0x00007fffffff70d0) at SizingManager.cc:295
    frame #11: 0x00007ffff49d8f96 libenergyplusapi.so.9.4.0`EnergyPlus::SimulationManager::ManageSimulation(state=0x00007fffffff70d0) at SimulationManager.cc:375
    frame #12: 0x00007ffff3b833b9 libenergyplusapi.so.9.4.0`RunEnergyPlus(state=0x00007fffffff70d0, filepath="\xc0p\xff\xff\xff\U0000007f"...) at EnergyPlusPgm.cc:401
    frame #13: 0x00007ffff3b82901 libenergyplusapi.so.9.4.0`EnergyPlusPgm(state=0x00007fffffff70d0, filepath="\xc0p\xff\xff\xff\U0000007f"...) at EnergyPlusPgm.cc:224
    frame #14: 0x00005555555e7154 energyplus`main(argc=1, argv=0x00007fffffffc218) at main.cc:60
    frame #15: 0x00007ffff147cb97 libc.so.6`__libc_start_main(main=(energyplus`main at main.cc:56), argc=1, argv=0x00007fffffffc218, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffc208) at libc-start.c:310
    frame #16: 0x00005555555e6f7a energyplus`_start + 42

(lldb) fr select 5
frame #5: 0x00007ffff4adff91 libenergyplusapi.so.9.4.0`EnergyPlus::SolarShading::ComputeWinShadeAbsorpFactors() at SolarShading.cc:11800
   11797	                        WindowShadingControl(WinShadeCtrlNum).ShadingType == WSC_ST_ExteriorShade ||
   11798	                        WindowShadingControl(WinShadeCtrlNum).ShadingType == WSC_ST_BetweenGlassShade) {
   11799	                        int ConstrNumSh = Surface(SurfNum).activeShadedConstruction; // Window construction number with shade
-> 11800	                        int TotLay = dataConstruction.Construct(ConstrNumSh).TotLayers; // Total layers in a construction
   11801	
   11802	
   11803	                        if (WindowShadingControl(WinShadeCtrlNum).ShadingType == WSC_ST_InteriorShade) {

(lldb) p Surface(SurfNum).Name
(std::__cxx11::string) $1 = "PUNCHED WINDOW 2"
(lldb) p Surface(SurfNum).activeShadedConstruction
(int) $2 = 0
```

**Comment by jmarrec** [comment]
From the Daylighting_School.idf

```
  FenestrationSurface:Detailed,
    Punched Window 2,        !- Name
    Window,                  !- Surface Type
    Vision Window,           !- Construction Name
    Classroom Daylighting Wall,  !- Building Surface Name
    ,                        !- Outside Boundary Condition Object
    AutoCalculate,           !- View Factor to Ground
    ,                        !- Frame and Divider Name
    1.0000,                  !- Multiplier
    4,                       !- Number of Vertices
    5.0292,9.1440,2.2860,  !- X,Y,Z ==> Vertex 1 {m}
    5.0292,9.1440,0.7620,  !- X,Y,Z ==> Vertex 2 {m}
    4.1148,9.1440,0.7620,  !- X,Y,Z ==> Vertex 3 {m}
    4.1148,9.1440,2.2860;  !- X,Y,Z ==> Vertex 4 {m}

  WindowShadingControl,
    Shade_control-view-Classroom,  !- Name
    Classroom,               !- Zone Name
    1,                       !- Shading Control Sequence Number
    InteriorShade,           !- Shading Type
    ,                        !- Construction with Shading Name
    AlwaysOn,                !- Shading Control Type
    ,                        !- Schedule Name
    ,                        !- Setpoint {W/m2, W or deg C}
    NO,                      !- Shading Control Is Scheduled
    NO,                      !- Glare Control Is Active
    MEDIUM REFLECT - MEDIUM TRANS SHADE,  !- Shading Device Material Name
    FixedSlatAngle,          !- Type of Slat Angle Control for Blinds
    ,                        !- Slat Angle Schedule Name
    ,                        !- Setpoint 2 {W/m2 or deg C}
    Classroom_DaylCtrl,      !- Daylighting Control Object Name
    Sequential,              !- Multiple Surface Control Type
    Punched Window 1,        !- Fenestration Surface 1 Name
    Punched Window 2,        !- Fenestration Surface 2 Name
    Punched Window 3;        !- Fenestration Surface 3 Name
```

**Comment by tijcolem** [comment]
Appears to be an issue in the cmake not getting the E+ package. 

e.g. on mac. 

```
Make Error at CMakeLists.txt:593 (file):
  file DOWNLOAD HASH mismatch

    for file: [/Users/jenkins/git/OpenStudioIncremental/develop/build/EnergyPlus-9.4.0-5a40ebd3b9-Darwin-macOS-10.13-x86_64.tar.gz]
      expected hash: [56add3a7c8e2290f3ba98e8316c781f9]
        actual hash: [d41d8cd98f00b204e9800998ecf8427e]
             status: [22;"HTTP response code said error"]

```

**Comment by tijcolem** [comment]
Looks like a simple file name issue. 

Cmake Link
https://github.com/NREL/EnergyPlus/releases/download/v9.4.0-RC1/EnergyPlus-9.4.0-5a40ebd3b9-Darwin-macOS-10.13-x86_64.tar.gz

Real Link
https://github.com/NREL/EnergyPlus/releases/download/v9.4.0-RC1/EnergyPlus-9.4.0-5a40ebd3b9-Darwin-macOS10.13-x86_64.tar.gz

**Comment by jmarrec** [comment]
The following tests FAILED:

* [ ] IlluminanceMapFixture.IlluminanceMapPlotMax (Failed)
* [ ] IlluminanceMapFixture.IlluminanceMapPlotMinMax (Failed)
* [ ] SqlFileFixture.AnnualTotalCosts (Failed)

**Comment by jmarrec** [comment]
I'm running the OpenStudio-resources tests with this branch right now.

**Comment by joseph-robertson** [comment]
> I'm running the OpenStudio-resources tests with this branch right now.

@jmarrec I'll run it against https://github.com/NREL/OpenStudio-resources/pull/114 and https://github.com/NREL/OpenStudio-resources/pull/115.

File: CMakeLists.txt [hunk 1]
```diff
@@ -530,37 +530,63 @@ endif()
...
+  # no-op
+  set(ENERGYPLUS_SYSTEM_VERSION "")
 endif()
 
 # EnergyPlus
 
+if((DEFINED ENERGYPLUS_EXE) AND NOT (ENERGYPLUS_EXE STREQUAL ""))
+  get_filename_component(OLD_ENERGYPLUS_EXTRACTED_DIR "${ENERGYPLUS_EXE}" DIRECTORY)
+  message("OLD_ENERGYPLUS_EXTRACTED_DIR=${OLD_ENERGYPLUS_EXTRACTED_DIR}")
+endif()
```
Comment (jmarrec) [inline CMakeLists.txt:565]: So this once merged to develop, should avoid the problem of FindEnergyPlus finding the wrong version (hopefully), coupled with the remove_directory call below.

File: CMakeLists.txt [hunk 2]
```diff
@@ -575,6 +601,9 @@ if(UNIX)
     endif()
 
     # Remove the old extracted dir if exists
+    if(DEFINED OLD_ENERGYPLUS_EXTRACTED_DIR)
+      execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory "${OLD_ENERGYPLUS_EXTRACTED_DIR}")
+    endif()
```
Comment (jmarrec) [inline CMakeLists.txt:606]: here

**Pull request merged**